### PR TITLE
Convert issue templates to forms

### PR DESCRIPTION
**Pull request by Semisol**
Converting issue templates to forms reduces the potential for new people to screw up issues, to then be marked as invalid and closed due to not adhering to the template which may deter new people.

<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [ ] (N/A) I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

### CODE REVIEW

**Comment by Semisol** [comment]
I confused assignees and reviewers 😅

**Review by nbsp — CHANGES_REQUESTED** [review]
"I have searched the archives and haven't found any related bug" should be added somewhere

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+#title: "[Bug] "
+labels: [bug]
+assignees: []
+body:
+  - type: markdown
+    attributes:
+      value: |
+        # Note
+        Issues are not a place to go ask support questions.
+        Please our [mailing list](https://lists.sr.ht/~tenacity/tenacity-discuss) and IRC (#tenacity on irc.libera.chat) ([web](https://web.libera.chat/#tenacity) | [matrix](https://matrix.to/#/#tenacity:libera.chat))
```
Comment (nbsp) [inline .github/ISSUE_TEMPLATE/bug_report.yaml]: That's a long line

```suggestion
        Please our [mailing list](https://lists.sr.ht/~tenacity/tenacity-discuss) \
        and IRC (#tenacity on irc.libera.chat) ([web](https://web.libera.chat/#tenacity) \ 
        | [matrix](https://matrix.to/#/#tenacity:libera.chat))

**Review by nbsp — DISMISSED** [review]

**Comment by nbsp** [comment]
Actually it doesn't work, wat
GitHub doesn't seem to respect YAML standards so whatever have the long line

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 1 above)
Comment (nbsp) [inline .github/ISSUE_TEMPLATE/bug_report.yaml]: > Please our

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+#title: "[Bug] "
+labels: [bug]
+assignees: []
+body:
+  - type: markdown
+    attributes:
+      value: |
+        # Note
+        Issues are not a place to go ask support questions.
+        Please ask support questions on the [mailing list](https://lists.sr.ht/~tenacity/tenacity-discuss) or IRC (#tenacity on irc.libera.chat) ([web](https://web.libera.chat/#tenacity) | [matrix](https://matrix.to/#/#tenacity:libera.chat))
```
Comment (nbsp) [inline .github/ISSUE_TEMPLATE/bug_report.yaml]: Matrix should be capitalized (also possibly Web, but that's less clear-cut)

**Review by nbsp — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        - macOS
+        - Windows 10
+        - Linux (Other, specify in version)
+        - Debian
+        - Fedora
+        - Arch
+        - Ubuntu
+        - ElementaryOS
+        - Linux Mint
+        - Manjaro
```
Comment (nbsp) [inline .github/ISSUE_TEMPLATE/bug_report.yaml]: 1. Should be sorted alphabetically
2. It should be fine to include only Debian/Arch/Fedora/Gentoo, and derivatives can be specified in version.
Either that or we just make this an input field

**Review by nbsp — CHANGES_REQUESTED** [review]

**Review by nbsp — APPROVED** [review]

**Review by emabrey — APPROVED** [review]

**Pull request merged**